### PR TITLE
MONGOID-4686 remove Document#to_a

### PR DESCRIPTION
**Pull request by Neilshweky**

### CODE REVIEW

**Review by p-mongo — APPROVED** [review]
This needs to be added to the release notes though.

**Comment by p-mongo** [comment]
And we should probably have a deprecation notice in 7.x.

File: docs/release-notes/mongoid-8.0.txt [hunk 1]
```diff
@@ -276,3 +276,9 @@ as well as ActiveRecord-compatible ``previously_new_record?`` and
 
   user.destroy
   user.previously_persisted?   # => true
+
+
+Removed Document#to_a method
+----------------------------
+
+The ``Document#to_a`` method has been removed in Mongoid 8.
```
Comment (Neilshweky) [inline docs/release-notes/mongoid-8.0.txt:284]: Does anything else need to be written here?

Comment (p-mongo) [reply docs/release-notes/mongoid-8.0.txt:284]: I think this is quite sufficient, thank you.

**Review by p-mongo — APPROVED** [review]

**Pull request merged**